src/java.desktop/unix/classes/sun/awt/X11/XFramePeer.java

```diff
@@ -80,6 +76,38 @@ void preInit(XCreateWindowParams params) {
                      Integer.valueOf(winAttr.decorations), Boolean.valueOf(winAttr.initialResizability),
                      Boolean.valueOf(!winAttr.nativeDecor), Integer.valueOf(winAttr.initialState));
         }
+
+        registerWindowDecorationChangeListener();
+    }
+
+    private void registerWindowDecorationChangeListener() {
+        if (target instanceof javax.swing.RootPaneContainer) {
```

Comment:
Probably this should be done via runtime check? Overwise the swing classes will be always loaded even if swing is not used. see the usage of SunToolkit.isInstanceOf in java.awt.WIndow.java:3965

Reply:
Thanks! Updated the code; please, have a look.